### PR TITLE
Guard against no service being submitted

### PR DESCRIPTION
**Pull request by ian-leggett**
## Description of change

When submitting a interaction with no service the form errors

## Test instructions
Add interaction without a service selected.
 
### Before 
Screenshot 2019-07-19 at 11 09 24

### After 
Screenshot 2019-07-19 at 11 37 57

### CODE REVIEW

File: test/unit/apps/interactions/transformers/interaction-form-body-to-api.test.js [hunk 1]
```diff
@@ -141,4 +141,23 @@ describe('#transformInteractionFormBodyToApiRequest', () => {
...
+      beforeEach(() => {
+        this.transformed = transformInteractionFormBodyToApiRequest(
+          {
+            service: undefined,
+          },
+          transformedServices
+        )
+      })
+
+      it('should pass empty string as service value', () => {
```
Comment (mcbhenwood) [inline test/unit/apps/interactions/transformers/interaction-form-body-to-api.test.js]: Maybe you should extend the test description to cover the empty `service_answers` too (or move the `service_answers` check to a separate text).

File: src/apps/interactions/transformers/interaction-form-body-to-api.js [hunk 2]
```diff
@@ -14,28 +14,31 @@ function transformInteractionFormBodyToApiRequest (props, services) {
...
-  serviceOptionStore.interactionQuestions.map(interactionQuestion => {
-    for (const [key, value] of Object.entries(props)) {
-      if (key === interactionQuestion.value) {
-        serviceAnswers[key] = {
-          [value]: {},
+  const serviceAnswers = {}
+  let service = ''
+  if (selectedServiceOption) {
+    const serviceHasSecondaryOptions = !!selectedServiceOption.secondaryOptions
+      .length
```
Comment (mcbhenwood) [inline src/apps/interactions/transformers/interaction-form-body-to-api.js]: I'm all for short lines but at first read this looks like we are casting `selectedServiceOption.secondaryOptions` to a boolean before checking the length property. I appreciate that doesn't make sense (and obviously on second glance isn't the case) but maybe the `selectedServiceOption.secondaryOptions.length` oughtn't be broken up in the case? Would reduce cognitive load when quickly parsing the code visually.

I'm not particularly wedded to this point; just observing.

File: test/unit/apps/interactions/transformers/interaction-form-body-to-api.test.js (hunk 1 above)
Comment (yeahfro) [reply test/unit/apps/interactions/transformers/interaction-form-body-to-api.test.js]: > move the service_answers check to a separate text

👍

Comment (ian-leggett) [reply test/unit/apps/interactions/transformers/interaction-form-body-to-api.test.js]: No worries I will split the test into two as Lee mentioned

File: src/apps/interactions/transformers/interaction-form-body-to-api.js (hunk 2 above)
Comment (ian-leggett) [reply src/apps/interactions/transformers/interaction-form-body-to-api.js]: Good point this does read better on one line too

**Pull request merged**